pkg/provider/nomad/nomad.go

```diff
@@ -235,55 +237,90 @@ func (p *Provider) loadConfiguration(ctx context.Context, configurationC chan<-
 }
 
 func (p *Provider) getNomadServiceData(ctx context.Context) ([]item, error) {
-	// first, get list of service stubs
-	opts := &api.QueryOptions{AllowStale: p.Stale}
-	opts = opts.WithContext(ctx)
+	jobsOpts := &api.QueryOptions{}
```

Comment:
```suggestion
	jobsOpts := &api.QueryOptions{AllowStale: p.Stale}
```

Reply:
WDYT of keeping the `getNomadServiceData` method at it was and creating a new method `getNomadServiceDataWithEmptyServices` and updating the `loadConfiguration` method liake this:

```go
func (p *Provider) loadConfiguration(ctx context.Context, configurationC chan<- dynamic.Message) error {
	var items []item
	var err error
	if p.AllowEmptyServices {
		items, err = p.getNomadServiceDataWithEmptyServices(ctx)
		if err != nil {
			return err
		}
	} else {
		items, err = p.getNomadServiceData(ctx)
		if err != nil {
			return err
		}
	}
	
	configurationC <- dynamic.Message{
		ProviderName:  p.name,
		Configuration: p.buildConfig(ctx, items),
	}

	return nil
}
```
It will avoid to current users that don't need empty services to be affected by the change of fetching Jobs instead of Services.
For large cluster it will impact memory consumption and if we fetching jobs instead of services.

Reply:
I agree, that would be a better approach. I'll make the suggested changes. Thanks!

Reply:
I implemented the suggested changes. Thanks!